core/flow/slot.go

```diff
@@ -85,9 +87,9 @@ func selectNodeByRelStrategy(rule *Rule, node base.StatNode) base.StatNode {
 func checkInLocal(tc *TrafficShapingController, resStat base.StatNode, batchCount uint32, flag int32) *base.TokenResult {
 	actual := selectNodeByRelStrategy(tc.rule, resStat)
 	if actual == nil {
-		logging.FrequentErrorOnce.Do(func() {
+		if ok := atomic.CompareAndSwapUint32(&logging.FrequentErrorOnce, 0, 1); ok {
```

Comment:
What is the difference between this modification and sync.once？

Reply:
`sync.Once` 
 this code block  are executed in two or more goroutines, only one can  execute , the others  must wait   it  finished  
 ,then, continue.  `sync.Once`    blocks the rest 

cas  is non-block